### PR TITLE
CUMULUS-2902: Removes dependency of aws-sdk from all packages

### PR DESCRIPTION
**Pull request by paulpilone**
**Summary:** Summary of changes

Addresses [CUMULUS-2902: D - (LAST) Remove all dependencies on AWS SDK v2](https://bugs.earthdata.nasa.gov/browse/CUMULUS-2902). This PR should be reviewed after https://github.com/nasa/cumulus/pull/3616 since it was originally branched from those changes.

## Changes

* Removes `aws-sdk` from all packages and replaces any usage of it with equivalent v3 syntax

## PR Checklist

- [x] Update CHANGELOG
- [x] Unit tests
- [x] Ad-hoc testing - Deploy changes and test manually
- [x] Integration tests

### CODE REVIEW

File: example/lambdas/ftpPopulateTestLambda/webpack.config.js
Comment (npauzenga) [inline example/lambdas/ftpPopulateTestLambda/webpack.config.js:30]: I should probably know this but what are we saying here? What does this change do?

File: packages/api/webpack.config.js
Comment (npauzenga) [inline packages/api/webpack.config.js:83]: Same question as above (same for all of the webpack configs). ELI5 please 😆

File: packages/aws-client/src/client.ts
Comment (npauzenga) [inline packages/aws-client/src/client.ts:53]: Presumably this was a V2 issue that V3 corrects?

File: packages/aws-client/src/client.ts
Comment (npauzenga) [inline packages/aws-client/src/client.ts:10]: What did this do?

File: packages/aws-client/src/lib/S3MultipartUploads.ts
Comment (npauzenga) [inline packages/aws-client/src/lib/S3MultipartUploads.ts:15]: Hmm maybe we can clean this up? It looks like we're importing `CompleteMultipartUploadOutput` as an alias, then exporting `CompleteMultipartUploadOutput` as an extension of that alias?

I guess our interface already shared a name with an AWS interface but the namespace was different. Kinda gross but I see why you did it this way. The grossness was pre-existing.

Maybe we rename our interface? Also maybe this doesn't really matter except that it's a little confusing at first glance.

File: packages/aws-client/src/types.ts
Comment (npauzenga) [inline packages/aws-client/src/types.ts:37]: Were we just not using these?

File: packages/tea-map-cache/webpack.config.js
Comment (npauzenga) [inline packages/tea-map-cache/webpack.config.js:14]: Should this have the `@`? Also what's up with the `@`? 😄

**Review by npauzenga — CHANGES_REQUESTED** [review]
This looks great @paulpilone! We were concerned about what might have been missed in this epic and it looks like you caught a few things. 

I had a couple clarifying questions but overall looks great.

File: packages/tea-map-cache/webpack.config.js
Comment (paulpilone) [reply packages/tea-map-cache/webpack.config.js]: @npauzenga As discussed in Slack this was originally changed because we want (in theory) to exclude @aws-sdk (v3) from our packages since it's provided in the Lambda environment for us.

However your question made me think about this more and it's a good thing it did. We're still targeting Node 16 which does _not_ include the v3 SDK in the Lambda runtime. So we can't exclude it.

Also, I'm not convinced that was excluding it the way we thought it was because most of our packages depend on `aws-client` which does not have a webpack configuration. I've added a comment to our Node upgrade ticket (https://bugs.earthdata.nasa.gov/browse/CUMULUS-3433) to make sure this gets re-added and in a way that actually does what we expect.

File: packages/aws-client/src/types.ts
Comment (paulpilone) [reply packages/aws-client/src/types.ts:37]: `AWS.Service` wasn't used and `AWS.DynamoDB.DocumentClient` became

```
export const dynamodbDocClient = (docClientOptions?: TranslateConfig, dynamoOptions?: object) =>
  DynamoDBDocument.from(
    awsClient(DynamoDB, '2012-08-10')(dynamoOptions),
    docClientOptions
  );
 ```

File: packages/aws-client/src/lib/S3MultipartUploads.ts
Comment (paulpilone) [reply packages/aws-client/src/lib/S3MultipartUploads.ts:15]: I added the alias because it was the path to least resistance but I think you're right and our interface should really be named something else. We already do something similar here: https://github.com/nasa/cumulus/pull/3618/files#diff-56edb8ddc2b0588b9a8b2f0bb5f33759f187a5bd46e30725d6e79cbe6ed46056L162 although I'm not sure us calling it `SQSMessage` is super clean either. 

Curious your opinion on if we should address both, this one, or neither in this PR.

File: packages/aws-client/src/client.ts
Comment (paulpilone) [reply packages/aws-client/src/client.ts:53]: Correct. Looks like that `serviceIdentifier` issue was a v2 bug.

File: packages/aws-client/src/client.ts
Comment (paulpilone) [reply packages/aws-client/src/client.ts:10]: AWS SDK v2 allowed you to configure the promise library it used and this is how you did it.

File: packages/aws-client/src/lib/S3MultipartUploads.ts
Comment (npauzenga) [reply packages/aws-client/src/lib/S3MultipartUploads.ts:15]: Yeah. What if we took a small step and instead of importing `CompleteMultipartUploadOutput` as an alias, we import it normally and rename _our_ interface `S3CompleteMultipartUploadOutput`?

At least that way we don't have something with exactly the same name as the AWS interface, which could(?) cause confusion.

File: packages/api/webpack.config.js
Comment (npauzenga) [reply packages/api/webpack.config.js]: Punting this conversation to https://github.com/nasa/cumulus/pull/3618#discussion_r1561655895

File: example/lambdas/ftpPopulateTestLambda/webpack.config.js
Comment (npauzenga) [reply example/lambdas/ftpPopulateTestLambda/webpack.config.js]: Punting this conversation to https://github.com/nasa/cumulus/pull/3618#discussion_r1561655895

File: packages/tea-map-cache/webpack.config.js
Comment (npauzenga) [reply packages/tea-map-cache/webpack.config.js]: Thanks for clarifying this. I think it makes sense to just remove the V2 external for now.

File: packages/aws-client/src/lib/S3MultipartUploads.ts
Comment (paulpilone) [reply packages/aws-client/src/lib/S3MultipartUploads.ts]: @npauzenga Pushed a small change that refactored that in this file. aws-client unit tests pass locally and waiting on CI now.

**Review by npauzenga — APPROVED** [review]
This looks great @paulpilone! Good to go pending CI.

**Pull request merged**
